python/vyos/ifconfig/interface.py

```diff
@@ -1102,9 +1102,10 @@ def update(self, config):
             self.del_addr('dhcp')
 
         # always ensure DHCPv6 client is stopped (when not configured as client
-        # for IPv6 address or prefix delegation
+        # for IPv6 address or prefix delegation)
         dhcpv6pd = dict_search('dhcpv6_options.pd', config)
-        if 'dhcpv6' not in new_addr or dhcpv6pd == None:
+        dhcpv6pd = dhcpv6pd != None and len(dhcpv6pd) != 0
```

Comment:
Writing `dhcpv6pd = dict_search('dhcpv6_options.pd', config) != None` should be sufficient

Reply:
To see why it is not sufficient:
```
vyos@vyos:~$ cat dhcpv6pd-debug.patch 
--- /usr/lib/python3/dist-packages/vyos/ifconfig/interface.py.orig	2020-08-26 19:07:24.000000000 +0000
+++ /usr/lib/python3/dist-packages/vyos/ifconfig/interface.py	2021-01-27 20:20:33.748000000 +0000
@@ -1104,6 +1104,8 @@
         # always ensure DHCPv6 client is stopped (when not configured as client
         # for IPv6 address or prefix delegation
         dhcpv6pd = dict_search('dhcpv6_options.pd', config)
+        with open('/tmp/dhcpv6pd', 'w') as f:
+            print(repr(dhcpv6pd), file=f)
         if 'dhcpv6' not in new_addr or dhcpv6pd == None:
             self.del_addr('dhcpv6')
 
vyos@vyos:~$ sudo patch -p0 <dhcpv6pd-debug.patch 
patching file /usr/lib/python3/dist-packages/vyos/ifconfig/interface.py
vyos@vyos:~$ sudo systemctl restart vyos-configd
vyos@vyos:~$ configure
[edit]
vyos@vyos# set interfaces ethernet eth0 dhcpv6-options 
[edit]
vyos@vyos# commit
[edit]
vyos@vyos# exit
Warning: configuration changes have not been saved.
exit
vyos@vyos:~$ cat /tmp/dhcpv6pd
{}
```

The empty `dict` does not equal `None`.